src/main/webapp/main/LandingPage.js

```diff
@@ -117,25 +123,29 @@ const LandingPage = hh(class LandingPage extends Component{
           });
         });
         this.setState(resp => {
+          this.props.hideSpinner();
           resp.taskList = taskListData;
           return resp;
         });
       }
-    }).catch(() => {});
+    } catch(e) {
+      this.props.hideSpinner();
+    }
   };
 
-  printComments = () => {
-
-  };
-
-  hasSession = () => {
-      User.isAuthenticated().then(resp => {
-        if (this._isMounted) {
-          this.setState({
-            logged: resp.data.session
-          })
-        }
-      }).catch(error => this.setState(() => { throw error; }));
+  hasSession = async () => {
+    try {
+      const resp = await User.isAuthenticated();
+      if (this._isMounted) {
+        this.setState({
+          logged: resp.data.session
+        })
+      }
+    } catch(error) {
+      this.setState(() => {
+        throw error;
+      }, () => this.props.hideSpinner());
```

Comment:
Does it make more sense to hide the spinner before setting error state?

Reply:
You're right, this has been modified.